### ISSUE TITLE
Systray Icon Not Loading

### ISSUE THREAD

**Issue by terminalpain** [safing/portmaster#1524]
**What happened**:

When starting up system and booting in to desktop (Linux Mint) there is no Portmaster icon in the systray.

**What did you expect to happen?**:

I expected the icon to be there.

**How did you reproduce it?**:

Reboot.

If I open the app via the shortcut in my Administration menu, shutdown portmaster, then reopen portmaster the icon re-appears.

Additional logs can be found here:
NJ{"Created":1714568703,"Modified":1714568703,"Expires":1717160703,"Deleted":0}S
executing notifier/portmaster-notifier version 0.3.6 on linux amd64

NJ{"Created":1714426580,"Modified":1714426580,"Expires":1717018580,"Deleted":0}S
executing notifier/portmaster-notifier version 0.3.6 on linux amd64
2024/04/29 23:20:04 systray error: failed to set IconPixmap prop: dbus: connection closed by user
2024/04/29 23:20:04 systray error: failed to update menu version: dbus: connection closed by user

NJ{"Created":1714429241,"Modified":1714429241,"Expires":1717021241,"Deleted":0}S
executing notifier/portmaster-notifier version 0.3.6 on linux amd64
2024/04/29 23:20:41 systray error: failed to register our icon with the notifier watcher (maybe no tray is running?): The name org.kde.StatusNotifierWatcher was not provided by any .service files
2024/04/29 23:50:48 systray error: failed to set IconPixmap prop: dbus: connection closed by user
2024/04/29 23:50:48 systray error: failed to update menu version: dbus: connection closed by user

NJ{"Created":1714471743,"Modified":1714471743,"Expires":1717063743,"Deleted":0}S
executing notifier/portmaster-notifier version 0.3.6 on linux amd64
2024/04/30 11:09:03 systray error: failed to register our icon with the notifier watcher (maybe no tray is running?): The name org.kde.StatusNotifierWatcher was not provided by any .service files
2024/04/30 12:36:41 systray error: failed to set IconPixmap prop: dbus: connection closed by user
2024/04/30 12:36:41 systray error: failed to update menu version: dbus: connection closed by user

These logs are from the notifier directory, there are plenty of them it seems.

**Comment by Raphty** [comment]
hey @terminalpain

If you are on Linux Mint I assume you are on Cinnamon? we do not support Cinnamon DE at the moment.
If you want to help us get support for Cinnamon please let us know.

**Comment by terminalpain** [comment]
Ye I am using Cinnamon DE, it didn't dawn on me that each DE would need extra work, but it makes sense, ah well no worries, it's just a little niggle (a tray icon). Thanks for the reply :)

**Comment by Raphty** [comment]
@terminalpain I can't promise anything, but we are working on replacing the UI backend with Tauri, and merging the notifier into the main UI. getting that to work on cinnamon might be easier.

**Comment by dlsensei** [comment]
Easyfix and not elegant to fix notifier not in systemtray in Linux mint 22 at statup.
just create 2 startup command in "startup applications"
1:
Stop notifier command:  
killall portmaster-notifier_v0-3-6   (check systemmonitor to get the correct name of notifier) 
add a delay of 10s for the startup command

2:
Start notifier command:  
/opt/safing/portmaster/portmaster-start notifier
add delay 20s

notifier at startup tadaaaa!!!

**Comment by terminalpain** [comment]
Thanks for the reply! I have moved on from Mint now though, and have been happily using CachyOS with Hyprland :)  Which unfortunately the same thing happens, it is running, but no systray icon.  I will try and mess with a delay though see if that helps, but I know if I refresh my waybar I lose it again. I actually had to do something similar with Strawberry because that had the habit of launching way before systray had a chance to start :)
